assets/scripts/lobby/chat4.js

```diff
@@ -359,7 +359,7 @@ function addToRoomChat(data) {
 
                 // console.log("true?"  + selectedChat[data[i].username]);
 
-                if (selectedChat[data[i].username] === true) {
+                if (selectedChat[data[i].username] === true && getIndexFromUsername(data[i].username)) {
```

Comment:
Please do something like `**getIndexFromUsername(data[i].username) !== undefined**`
getIndexFromUsername() can return the number 0. When it does, it will return a False and make the expression False. Please test this.

Reply:
(I know this because I once spent hours debugging and this was the exact cause back then!)